src/compas/numerical/dr/dr_numpy.py

```diff
@@ -116,12 +116,19 @@ def dr_numpy(vertices, edges, fixed, loads, qpre,
     # --------------------------------------------------------------------------
     # input processing
     # --------------------------------------------------------------------------
-    qpre = qpre or [0.0 for _ in range(num_e)]
-    fpre = fpre or [0.0 for _ in range(num_e)]
-    lpre = lpre or [0.0 for _ in range(num_e)]
-    linit = linit or [0.0 for _ in range(num_e)]
-    E = E or [0.0 for _ in range(num_e)]
-    radius = radius or [0.0 for _ in range(num_e)]
+
+    def init_array(array, length):
+        if array is None or (isinstance(array, list) and len(array) == 0):
+            return zeros((length,), dtype=float)
+        else:
+            return array
+
+    qpre = init_array(qpre, num_e)
```

Comment:
@brgcode `not qpre` behind `or` will still trigger  evaluation on np array inputs thus throws the error. How about this?

Reply:
you don't have to check for `list`, `len(array)` works both for `numpy` arrays and python lists

Reply:
the `else` is also unnecessary :)

Reply:
@brgcode right.... makes sense. updated!